File: 프로그래머스/0/181856. 배열 비교하기/배열 비교하기.java
class Solution {
    public int solution(int[] arr1, int[] arr2) {
        if (arr1.length != arr2.length) {
            return arr1.length > arr2.length ? 1 : -1;
        }
        
        int sum1 = 0;
        for (int x : arr1) {
            sum1 += x;
        }
        
        int sum2 = 0;
        for (int x : arr2) {
            sum2 += x;
        }
        
        if (sum1 > sum2) {
            return 1;
        } else if (sum1 < sum2) {
            return -1;
        }
        
        return 0;
    }
}

File: 프로그래머스/0/181856. 배열 비교하기/배열 비교하기.kt
class Solution {
    fun solution(arr1: IntArray, arr2: IntArray): Int {
        if (arr1.size != arr2.size) {
            return if (arr1.size > arr2.size) 1 else -1
        }

        val sum1 = arr1.sum()
        val sum2 = arr2.sum()

        return when {
            sum1 > sum2 -> 1
            sum1 < sum2 -> -1
            else -> 0
        }
    }
}
